target/snitch_cluster/sw/snax/streamer-simd/src/snax-streamer-simd-lib.c

```diff
@@ -0,0 +1,162 @@
+// Copyright 2023 KU Leuven.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+//
+// Xiaoling Yi <xiaoling.yi@esat.kuleuven.be>
+
+#include "snax-streamer-simd-lib.h"
+
+int32_t gen_csr0_config(uint8_t input_zp_i, uint8_t output_zp_i,
+                        uint8_t shift_i, uint8_t max_int_i) {
+    // encode the configuration into a single 32-bit integer
+    return ((int32_t)max_int_i << 24) | ((int32_t)shift_i << 16) |
+           ((int32_t)output_zp_i << 8) | (int32_t)input_zp_i;
+}
+
+int32_t gen_csr1_config(uint8_t min_int_i, bool double_round_i) {
+    // encode the configuration into a single 32-bit integer
+    return ((uint8_t)double_round_i << 8) | (uint8_t)min_int_i;
+}
+
+int32_t gen_csr2_config(uint32_t multiplier_i) { return multiplier_i; }
```

Comment:
Yeah, I think they better have more appropriate name I guess? XD I dunno. Something like change `gen_csr0_config` to `data_offset`. Then the `gen_csr1_config` is like `data_shift` or something? Just a suggestion but we can keep if it it's not that important.

Reply:
No, `gen_csr0_config` is not purely `data_offset`! Let's remain this for now!